src/closure.js

```diff
@@ -5,17 +5,93 @@ const counter = () => {
   // Example: const newCounter = counter();
   // newCounter(); // 1
   // newCounter(); // 2
+
+  // version 1 √
+  // let value = 0;
+  // const increment = () => value += 1;
+  // return increment;
+
+  // verion 2 √
+  // let value = 0;
+  // const increment = i => value += 1; // Interesting that this works - why?
+  // return increment;
+
+  // version 3 √
+  let value = 0;
+  return () => value += 1;
 };
 
 const counterFactory = () => {
   // Return an object that has two methods called `increment` and `decrement`.
   // `increment` should increment a counter variable in closure scope and return it.
   // `decrement` should decrement the counter variable and return it.
+
+  // version 1 √
+  let value = 0;
+  const obj = {
+    increment() { return value += 1; },
+    decrement() { return value -= 1; }
+  };
+  return obj;
+
+  // version 2 √
+  // let value = 0;
+  // return {
+  //   increment() { return value += 1; },
+  //   decrement() { return value -= 1; }
+  // };
+
+  // version 3 √
+  // let value = 0;
+  // return {
+  //   increment: () => value += 1,
+  //   decrement: () => value -= 1
+  // };
 };
 
 const limitFunctionCallCount = (cb, n) => {
   // Should return a function that invokes `cb`.
   // The returned function should only allow `cb` to be invoked `n` times.
+  //
+  // Ely and Sarah helped me in Slack, but i do not understand what's going on here
+  // I got the limitFuncCall example from the JS course with sayHi... but still not solid on my understanding
+
+  // version 1 √
+  let limit = 0;
+  return (...args) => {
+    limit++;
+    if (n > limit) {
+      // limit++; // <--- acts the same as limit++ before if eval
+      // console.log(limit);
+      return cb(...args);
+    }
+    return null; // <--- still not sure why we need to return null, just because the test is set up that way?
```

Comment:
Yeah, we want to have something specified to return if the function has reached its limit.